### DISCUSSION TITLE
Concourse affected by CVE-2021-45046 (Apache Log4j Security Vulnerability)?

### DISCUSSION THREAD

**Discussion by andresm6** [concourse/concourse#7887] [Help & Support] [answered]
Hello, 

I hope I have the right category and and format for this. Apologies if not.

I am doing some research for my company and wanted to get validation that Concourse is not affected by the recent Log4j vulnerability - [CVE-2021-45046](https://logging.apache.org/log4j/2.x/security.html). 

Based on my understanding, it looks like Concourse is written in go, but due to the circumstances surrounding the CVE and how many applications it has affected, I wanted to make sure I wasn't missing a module somewhere that may be using the log4j library.

Any help would be appreciated.

**Comment by skreddy6673** [comment]
@taylorsilva Do you have any update on this, like the initial comment mentioned Concourse is written in Golang for logging, but just want a confirmation on this.

Thanks in Advance!

**Comment by taylorsilva** [comment, marked as answer]
Concourse itself does not use Log4j. This includes the `concourse`, `runc`, `gdn`, and `containerd` binaries. None of the resource types that come bundled with Concourse are vulnerable either. You can find that list of **Bundled Resource Types** under any release: https://github.com/concourse/concourse/releases

From the main repo: https://github.com/concourse/concourse
<img width="313" alt="image" src="https://user-images.githubusercontent.com/4583837/146399741-0d8c0b1e-f349-42f9-91c9-3ca25d9ed281.png">
There's zero Java in our code.
(Javascript != Java)

For logging we use this Go library: https://github.com/cloudfoundry/lager/
Which is 100% written in Go.

**Reply by skreddy6673** [reply]
Thank you Taylor..

**Reply by andresm6** [reply]
great, thank you for the response!
